src/main/java/com/strandls/utility/service/impl/UtilityServiceImpl.java

```diff
@@ -452,15 +485,63 @@ public HomePageData getHomePageData(HttpServletRequest request, Boolean adminLis
 
 	}
 
+	private Map<String, Map<Long, List<GallerySlider>>> groupGallerySliders(List<GallerySlider> galleryData) {
+		Map<String, Map<Long, List<GallerySlider>>> grouped = new HashMap<>();
+		try {
+			for (GallerySlider gallery : galleryData) {
```

Comment:
instead of making individual requests to user api in a for loop, you can use `/v1/user/ibp/userList` from user service to fetch everything in a single api call.